Dockerfile

```diff
@@ -14,9 +16,13 @@ COPY main.go main.go
 COPY api/ api/
 COPY internal/ internal/
 
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
 ENV CGO_ENABLED=0
-ENV GOOS=linux
-ENV GOARCH=amd64
+ENV GOOS=${TARGETOS:-linux}
+ENV GOARCH=${TARGETARCH}
```

Comment:
In the default file these are set inline with the `go build command`, but I kept the style of the existing file.